Refresh.GameServer/Types/Levels/Categories/LevelCategory.cs

```diff
@@ -3,86 +3,36 @@
     
-    #if DEBUG
-    private static readonly List<string> ExpectedParameters = new()
-    {
-        "count",
-        "skip",
-        "gameVersion",
-    };
-    #endif
-    
-    internal LevelCategory(string apiRoute, string gameRoute, bool requiresUser, string funcName) : this(apiRoute, new []{gameRoute}, requiresUser, funcName) {}
+    internal LevelCategory(string apiRoute, string gameRoute, bool requiresUser) : this(apiRoute, new []{gameRoute}, requiresUser) {}
     
-    internal LevelCategory(string apiRoute, string[] gameRoutes, bool requiresUser, string funcName)
+    internal LevelCategory(string apiRoute, string[] gameRoutes, bool requiresUser)
     {
         this.ApiRoute = apiRoute;
         this.GameRoutes = gameRoutes;
         
         this.RequiresUser = requiresUser;
-
-        MethodInfo? method = Methods.Value.FirstOrDefault(m => m.Name == funcName);
-        if (method == null) throw new ArgumentNullException(nameof(funcName), 
-            $"{nameof(funcName)} must point to a method on {nameof(GameDatabaseContext)}! Use nameof() to assist with this.");
-
-        #if DEBUG
-        List<string> parameters = method.GetParameters().Select(p => p.Name).ToList()!;
-        foreach (string expectedParameter in ExpectedParameters)
-        {
-            if (parameters.Contains(expectedParameter)) continue;
-            throw new InvalidOperationException($"Cannot bind to {funcName}() when it is missing a {expectedParameter} parameter");
-        }
-        #endif
-
-        this._method = method;
     }
 
     [JsonProperty] public readonly string ApiRoute;
     public readonly string[] GameRoutes;
     
     [JsonProperty] public readonly bool RequiresUser;
-    private readonly MethodInfo _method;
 
     [Pure]
-    public virtual DatabaseList<GameLevel>? Fetch(RequestContext context, int skip, int count, MatchService matchService, GameDatabaseContext database, GameUser? user, TokenGame gameVersion, object[]? extraArgs = null)
-    {
-        if (this.RequiresUser && user == null) return null;
-
-        IEnumerable<object> args;
-
-        // ReSharper disable once ConvertIfStatementToConditionalTernaryExpression
-        if (this.RequiresUser)
-#pragma warning disable CS8601
-            args = new object[] { user, count, skip, gameVersion };
-#pragma warning restore CS8601
-        else
-            args = new object[] { count, skip, gameVersion };
-
-        if (extraArgs != null) args = args.Concat(extraArgs);
-
-        try
-        {
-            return (DatabaseList<GameLevel>)this._method.Invoke(database, args.ToArray())!;
-        }
-        catch
-        {
-            Console.WriteLine(this.ApiRoute);
-            throw;
-        }
-    }
+    public abstract DatabaseList<GameLevel>? Fetch(RequestContext context, int skip, int count, MatchService matchService, GameDatabaseContext database, GameUser? user, TokenGame gameVersion,
+        LevelFilterSettings levelFilterSettings);
```

Comment:
If we're gonna be passing a LevelFilterSettings object everywhere, we might as well move some stuff to it like `gameVersion`. This method's parameter list is very very long. Though, maybe not `skip` and `count` since that would mean it would be pretty annoying to use this API outside of the context of a request